File: Demo/ObjcDemo/NetImageController.m
//
//  NetImageController.m
//  ObjcDemo
//
//  Created by kelin on 2020/11/10.
//

#import "NetImageController.h"
#import "KKAutoCycleScrollView.h"
#import <SDWebImage/SDWebImage.h>

@interface NetImageController ()<KKAutoCycleScrollViewDelegate, KKAutoCycleScrollViewDataSource>
@property (nonatomic, strong) KKAutoCycleScrollView *cycleView;
@end

@implementation NetImageController

NSArray *netImages() {
    return @[
        @"https://icweiliimg6.pstatp.com/weili/l/378712400698146835.jpg",
        @"https://icweiliimg6.pstatp.com/weili/l/903360033333313563.jpg",
        @"https://weiliicimg1.pstatp.com/weili/l/812099039242354699.jpg",
        @"https://icweiliimg1.pstatp.com/weili/l/812022923831934995.jpg"
    ];
}
- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    self.title = @"网络图片加载";
    [self initCycleView];
}

- (void)initCycleView {
    _cycleView = [[KKAutoCycleScrollView alloc] initWithFrame:CGRectMake(0, 100, self.view.frame.size.width, 200)];
    _cycleView.delegate = self;
    _cycleView.dataSource = self;
    _cycleView.duration = 2;
    [_cycleView create];
    
    [self.view addSubview:_cycleView];
}

- (void)cycleView:(KKAutoCycleScrollView *)cycleView addSubViewsInPageView:(UIView *)pageView subViewAtIndex:(NSInteger)index {
    UIImageView *imageView = [[UIImageView alloc] initWithFrame:pageView.bounds];
    [imageView sd_setImageWithURL:[NSURL URLWithString:netImages()[index]]];
    [pageView addSubview:imageView];
}

- (NSInteger)numberOfViewCountInCycleView:(KKAutoCycleScrollView *)cycleView {
    return netImages().count;
}

- (void)cycleView:(KKAutoCycleScrollView *)cycleView didClickPageViewAtIndex:(NSInteger)index {
    NSLog(@"current page is %ld", index);
}

- (void)dealloc {
    [_cycleView destory];
}
@end

File: Demo/SwiftDemo/NetImageController.swift
//
//  NetImageController.swift
//  SwiftDemo
//
//  Created by kelin on 2020/11/10.
//

import Foundation
import UIKit

import SDWebImage

class NetImageController: UIViewController, KKAutoCycleScrollViewDelegate, KKAutoCycleScrollViewDataSource {
    
    var cycleView:KKAutoCycleScrollView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "网络图片加载"
        self.view.backgroundColor = UIColor.white
        
        self.initCycleView()
    }
    
    func initCycleView() -> Void {

        self.cycleView = KKAutoCycleScrollView(frame: CGRect(x: 0, y: 100, width: self.view.frame.size.width, height: 200))
        self.cycleView.delegate = self;
        self.cycleView.dataSource = self;
        self.cycleView.duration = 2
        self.cycleView.create()
        
        self.view.addSubview(self.cycleView)
    }
    
    func numberOfViewCountInCycleView(cycle: KKAutoCycleScrollView) -> NSInteger {
        return TestData.netImages().count
    }
    
    func cycleViewAddSubViewsInPageViewAtIndex(cycle: KKAutoCycleScrollView, pageView: UIView, index: NSInteger) {
        let imageView = UIImageView(frame: pageView.bounds)
        imageView.sd_setImage(with: URL(string: TestData.netImages()[index]), completed: nil)
        pageView.addSubview(imageView)
    }
    
    func cycleViewDidClickPageViewAtIndex(cycleView: KKAutoCycleScrollView, index: NSInteger) {
        print("current page is %ld", index)
    }
    
    deinit {
        self.cycleView.destory()
    }
}

